### PR TITLE
small changes to the loadthread

### PR DESCRIPTION
**Pull request by jgrewe**
experienced segfaults on my computer. These changes fixed them for now

### CODE REVIEW

File: plotter/lineplotter.cpp [hunk 1]
```diff
@@ -408,35 +408,25 @@ void LinePlotter::testThreads(QCPRange range) {
...
     for(int i=0; i<arrays.size(); i++) {
         int xDim = guess_best_xdim(arrays[i]);
         QCPGraph *graph = ui->plot->graph(graphIndex);
 
+        if(arrays[i].dataExtent().size() == 1) {
+            graphIndex += 1;
+        } else if(arrays[i].dataExtent().size() == 2) {
+            graphIndex += arrays[i].dataExtent()[2-xDim];
+        }
         if(graph->dataCount() == 0) {
```
Comment (Ott-Alexander) [inline plotter/lineplotter.cpp:422]: As far as i know this shouldn't lead to seq faults. The access with dataExtent().size() was fine in my tests and it's still there.

File: utils/loadthread.cpp [hunk 2]
```diff
@@ -166,8 +164,8 @@ void LoadThread::load2D(nix::DataArray array, nix::NDSize start, nix::NDSize ext
             }
             start[xDimIndex] = offset + i * chunksize;
 
-            array.getData(array.dataType(),chunkdata.data(),extent, start);
-
+            //array.getData(array.dataType(),chunkdata.data(),extent, start);
+            array.getData(chunkdata, extent, start);
             loadedData.insert(loadedData.end(), chunkdata.begin(), chunkdata.end());
```
Comment (Ott-Alexander) [inline utils/loadthread.cpp:168]: This is what i'm expecting to have caused the segfaults at least without actually testing it.

Comment (jgrewe) [reply utils/loadthread.cpp:168]: yes, that was my first impression as well, but the data that was loaded is 1D, so this function was never called

File: plotter/lineplotter.cpp (hunk 1 above)
Comment (jgrewe) [reply plotter/lineplotter.cpp:422]: it is not the call itself but the concurrent access, methinks

might strongly depend on the machine, worked on my linux computer but crached on mac

**Pull request merged**